src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -59,24 +57,30 @@ static zns_sm_metadata_t make_metadata(
   
   auto config_segment_size = get_conf<Option::size_t>(
     "seastore_segment_size");
-  logger().error("CONFIG SIZE: {}", config_segment_size);
-  size_t zones_per_segment = config_segment_size / zone_capacity;
-  
+
+  size_t segment_size = (zone_size == 0) ? config_segment_size : (zone_size << SECT_SHIFT);
+  size_t zones_per_segment = segment_size / (zone_size << SECT_SHIFT);
   size_t segments = (num_zones - 1) * zones_per_segment;
   
-  logger().debug(
-    "{}: size {}, block_size {}, allocated_size {}, configured_size {}, "
-    "segment_size {}",
-    __func__,
-    data.size,
+  INFO(
+    "\n\t device size {}, block_size {}, allocated_size {}, config_size {},"
+    "\n\t config_segment_size {}, total zones {}, zone_size {}, zone_capacity {},"
+    "\n\t total segments {}, zones per segment {}, segment size {}",
```

Comment:
I think there's no need to add new lines here.

Reply:
Ok.